exchanges/kucoin/kucoin.go

```diff
@@ -311,352 +205,213 @@ func (ku *Kucoin) GetKlines(ctx context.Context, pair, period string, start, end
 	params := url.Values{}
 	if base != "" {
 		params.Set("base", base)
 	}
 	if currencies != "" {
 		params.Set("currencies", currencies)
 	}
-	var resp map[string]string
-	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, common.EncodeURLValues(kucoinGetFiatPrice, params), &resp)
+	var resp map[string]types.Number
+	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, fiatPriceEPL, common.EncodeURLValues("/v1/prices", params), &resp)
+}
+
+// GetLeveragedTokenInfo returns leveraged token information
+func (ku *Kucoin) GetLeveragedTokenInfo(ctx context.Context, ccy currency.Code) ([]LeveragedTokenInfo, error) {
+	params := url.Values{}
+	if !ccy.IsEmpty() {
+		params.Set("currency", ccy.String())
+	}
+	var resp []LeveragedTokenInfo
+	return resp, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, leveragedTokenInfoEPL, http.MethodGet, common.EncodeURLValues("/v3/etf/info", params), nil, &resp)
 }
 
 // GetMarkPrice gets index price of the specified pair
-func (ku *Kucoin) GetMarkPrice(ctx context.Context, pair string) (*MarkPrice, error) {
-	if pair == "" {
-		return nil, currency.ErrCurrencyPairEmpty
+func (ku *Kucoin) GetMarkPrice(ctx context.Context, symbol string) (*MarkPrice, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
 	}
 	var resp *MarkPrice
-	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, fmt.Sprintf(kucoinGetMarkPrice, pair), &resp)
+	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, getMarkPriceEPL, "/v1/mark-price/"+symbol+"/current", &resp)
 }
 
 // GetMarginConfiguration gets configure info of the margin
 func (ku *Kucoin) GetMarginConfiguration(ctx context.Context) (*MarginConfiguration, error) {
 	var resp *MarginConfiguration
-	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, kucoinGetMarginConfiguration, &resp)
+	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, getMarginConfigurationEPL, "/v1/margin/config", &resp)
 }
 
 // GetMarginAccount gets configure info of the margin
 func (ku *Kucoin) GetMarginAccount(ctx context.Context) (*MarginAccounts, error) {
 	var resp *MarginAccounts
-	return resp, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, http.MethodGet, kucoinGetMarginAccount, nil, &resp)
+	return resp, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, marginAccountDetailEPL, http.MethodGet, "/v1/margin/account", nil, &resp)
 }
 
-// GetMarginRiskLimit gets cross/isolated margin risk limit, default model is cross margin
-func (ku *Kucoin) GetMarginRiskLimit(ctx context.Context, marginModel string) ([]MarginRiskLimit, error) {
+// GetCrossIsolatedMarginRiskLimitCurrencyConfig risk limit and currency configuration of cross margin/isolated margin.
+// isIsolated: true - isolated, false - cross ; default false
+func (ku *Kucoin) GetCrossIsolatedMarginRiskLimitCurrencyConfig(ctx context.Context, isIsolated bool, symbol string, ccy currency.Code) ([]RiskLimitCurrencyConfig, error) {
```

Comment:
Your response type is not adding fields from the response section: https://www.kucoin.com/docs/rest/margin-trading/margin-info/get-cross-isolated-margin-risk-limit-currency-config
Please, when creating response types, check the json _and_ the API documents for fields.

Given that I have seen this a couple of times now, I will leave my review here and request that you go over the newly added v3 endpoints, and ensure that all response data is accounted for